File: Go/Top Interview Questions/TwoSum.go
package main

import (
	"fmt"
)

func twoSum(nums []int, target int) [2]int{
	ans := [2]int{0,0};
	for index, num := range nums {
		diff := target - num
		for i:= index+1; i<len(nums); i++ {
			if nums[i] == diff{
				ans[0] = index;
				ans[1] = i;

				return ans;
			}
		}
	}
	 return ans;
}

func main() {
	nums := []int{3,3};
	var target int = 6;
	fmt.Println(twoSum(nums,target))
}

File: Java/Top Interview Questions/TwoSum.java
public class TwoSum {
  public static void main(String[] args) {
    int[] nums = new int[] { 1, 2, 3, 4 };
    int target = 5;
    int[] ans = twoSum(nums, target);
    System.out.println("[" + ans[0] + "," + ans[1] + "]");
  }

  public static int[] twoSum(int[] nums, int target) {
    int diff;
    int[] result = new int[] { 0, 0 };

    for (int i = 0; i < nums.length; i++) {
      diff = target - nums[i];
      for (int j = i + 1; j < nums.length; j++) {
        if (diff == nums[j]) {
          result[0] = i;
          result[1] = j;

          return result;
        }
      }
    }
    return result;
  }

}
